### DISCUSSION TITLE
Brew for macOS Sequoia

### DISCUSSION THREAD

**Discussion by rvalue** [Homebrew/discussions#5643] [Getting started] [answered]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.3.21-4-gfd7f2bb
ORIGIN: https://github.com/Homebrew/brew
HEAD: fd7f2bb3b17db8b0f3a50b39ddcd145889507dd5
Last commit: 2 weeks ago
Core tap HEAD: ccb7b6b462b4be1bf8200dd6b693abdbe8a83d8f
Core tap last commit: 5 weeks ago
Core tap branch: fix-flex-pc
Core tap JSON: 12 Sep 19:04 UTC
Core cask tap HEAD: 730d44f4edad75035f6c5a9bd51e89271af4b79e
Core cask tap last commit: 5 weeks ago
Core cask tap JSON: 12 Sep 19:04 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_DISPLAY: /private/tmp/com.apple.launchd.jnHjKtUSWl/org.xquartz:0
HOMEBREW_MAKE_JOBS: 8
HOMEBREW_SORBET_RUNTIME: set
Homebrew Ruby: 3.3.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.4_1/bin/ruby
CPU: octa-core 64-bit arm_firestorm_icestorm
Clang: 16.0.0 build 1600
Git: 2.39.5 => /Applications/Xcode.app/Contents/Developer/usr/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 14.6.1-arm64
CLT: 15.3.0.0.1.1708646388
Xcode: 16.0
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

What is the general recommendation of things to do before & after a macOS upgrade to make sure brew packages work fine?

Is there a manual or automated way to find in advance what formulas are not yet supported with macOS Sequoia? For example, may be the package is supported on the new macOS but there needs to be some formula update for its installation/upgradation to work with brew.

How does brew behave on a macOS version it does not know about?

**Comment by SMillerDev** [comment, marked as answer]
> What is the general recommendation of things to do before & after a macOS upgrade to make sure brew packages work fine?

Running a `brew upgrade` is never a bad idea after an update.

> Is there a manual or automated way to find in advance what formulas are not yet supported with macOS Sequoia?

You could check on formulae.brew.sh if it has a bottle. I'm fairly certain anything that worked on 14.x works on 15.x though.

> For example, may be the package is supported on the new macOS but there needs to be some formula update for its installation/upgradation to work with brew.

If the software works it'll work in brew.

> How does brew behave on a macOS version it does not know about?

It'll just install for an older version, but it knows about all versions that are public (even in developer beta).

**Comment by BigBobDalton** [comment]
My brew will not upgrade to Sequoia only stuck in Sonoma.  I dunno why, guess it doesn't matter since it works. 😂

**Reply by adah1972** [reply]
Does Sequoia have separate bottles from Sonoma? I upgraded from Ventura to Sequoia, and the new bottles are all Sonoma.
